solveengine/mipmodel.py

```diff
@@ -648,3 +644,73 @@ def __imul__(self, other):
     
     def __str__(self):
         return "".join([self._name, " : ", str(self._value)])
+
+
+def _check_matrices(f, A, b, Aeq, beq, lb, ub, int_list, bin_list):
+    """Check that the dimensions of the matrices match with each other
+
+    Complete the vectors lb, ub, int_list, bin_list
+    with values by default if they are shorter than the number of variables
+    """
+    check_instance(fct_name='build_with_matrices', value=f,
+                   name='f', type_=(list, ndarray))
+    check_instance(fct_name='build_with_matrices', value=A,
+                   name='A', type_=ndarray)
+    check_instance(fct_name='build_with_matrices', value=b,
+                   name='b', type_=list)
+
+    nb_vars = len(f)
+    if len(A.shape) != 2:
```

Comment:
I do not see why we restrict this to be ndarrays?
we can achieve what we want using the methods:
A[x][y] 
len(A)
len(A[x])
which is supported by ndarray and list of lists
it can also be easily supported by any custom class which calculates these values on the fly
hence I would not type check this and any of the other inputs

the pythonic programming style determines an object's type by inspection of its method or attribute signature rather than by explicit relationship to some type object. Typecheck, for me, only make sense for custom/local types.